### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by ardeshirj**
Initial pull request.

This is the initial project including the user authentication. We gonna use this as an example of code review.

### CODE REVIEW

**Comment by mindaika** [comment]
Test comment. Please ignore.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# See https://help.github.com/articles/ignoring-files for more about ignoring files.
+#
+# If you find yourself ignoring temporary files generated by your text editor
+# or operating system, you probably want to add a global ignore instead:
+#   git config --global core.excludesfile '~/.gitignore_global'
+
```
Comment (RadVenture) [inline .gitignore]: WHY IS THIS NEWLINE HERE?!? @ardeshirj

**Comment by RadVenture** [comment]
@ardeshirj Looks good to me!
@sector46 
@lewisric 
@beazley2 
Can you guys take a look as well?

**Comment by beazley2** [comment]
@lewisric @ardeshirj Fine by me as well

**Comment by CATcommander** [comment]
Sweet.

**Pull request merged**